scan/lib/scan/detect_values.rb

```diff
@@ -137,26 +172,23 @@ def self.detect_simulator(devices, requested_os_type, deployment_target_key, def
       # We create 2 lambdas, which we iterate over later on
       # If the first lambda `matches` found a simulator to use
       # we'll never call the second one
-
       matches = lambda do
         set_of_simulators = devices.inject(
           Set.new # of simulators
         ) do |set, device_string|
           pieces = device_string.split(regular_expression_for_split_on_whitespace_followed_by_parenthesized_version)
 
-          selector = ->(sim) { pieces.count > 0 && sim.name == pieces.first }
-
           display_device = "'#{device_string}'"
 
           set + (
             if pieces.count == 0
               [] # empty array
             elsif pieces.count == 1
               simulators
-                .select(&selector)
+                .select { |sim| sim.name == pieces.first && compatible_with_sdk(sim) }
```

Comment:
would it make sense to introduce a `highest_compatible_simulator(name)` method and reuse it both here and line 220? It would do a .last instead of pop(1) here

We could reuse it like 

```
[ highest_compatible_simulator(pieces.first) ]
```

here and 

```
[ highest_compatible_simulator(default_device_name) || simulators.first ] 
```
below.